x-pack/plugins/apm/public/hooks/use_transaction_latency_correlations_fetcher.ts

```diff
@@ -48,16 +41,42 @@ export const useCorrelations = (params: CorrelationsOptions) => {
   const [isComplete, setIsComplete] = useState(false);
   const [isRunning, setIsRunning] = useState(false);
   const [loaded, setLoaded] = useState<number>(0);
-  const [rawResponse, setRawResponse] = useState<RawResponse>();
+
+  const [ccsWarning, setCcsWarning] = useState<RawResponse['ccsWarning']>(
+    false
+  );
+  const [histograms, setHistograms] = useState<RawResponse['values']>([]);
+  const [log, setLog] = useState<RawResponse['log']>([]);
+  const [overallHistogram, setOverallHistogram] = useState<
+    RawResponse['overallHistogram']
+  >();
+  const [percentileThresholdValue, setPercentileThresholdValue] = useState<
+    RawResponse['percentileThresholdValue']
+  >();
+
   const [timeTook, setTimeTook] = useState<number | undefined>();
   const [total, setTotal] = useState<number>(100);
   const abortCtrl = useRef(new AbortController());
   const searchSubscription$ = useRef<Subscription>();
 
   function setResponse(response: IKibanaSearchResponse<RawResponse>) {
-    // @TODO: optimize rawResponse.overallHistogram if histogram is the same
     setIsRunning(response.isRunning || false);
-    setRawResponse(response.rawResponse);
+    setCcsWarning(response.rawResponse?.ccsWarning ?? false);
+    setHistograms(response.rawResponse?.values ?? []);
+    setLog(response.rawResponse?.log ?? []);
```

Comment:
Same as above. Why a different state handler for each? If we want to cherry-pick certain props we can do

```ts
setResponse({
  ccsWarning: response.rawResponse?.ccsWarning ?? false,
  histogram: response.rawResponse?.values ?? [],
  ...
})

Reply:
Originally we just had a single `setRawResponse` (see https://github.com/elastic/kibana/commit/2b361905381cca22372039fc9fc0f1c2aa5b23cf#diff-e78a6067e533c365dc4359166ad8c11d74df73448ffffb7c071f4c2e09aca969L51). It made it difficult to manage the loading behavior of the chart, because a refresh would trigger a reset and empty the chart and not an in-place update.

Since we moved some parts of the response to single state handlers it seemed counter-intuitive to still have a `[response, setResponse]` state handler around which didn't handle the full response.

We could return to use a single `setResponse` but then it would need to be set with a callback considering the previous state to handle the update and avoid the full chart reset. Let me know what you prefer.